modules/charmed-temporal/variables.tf

```diff
@@ -0,0 +1,61 @@
+
+variable "model" {
+  description = "Reference to an existing model resource or data source for the model to deploy to."
+  type        = string
+}
+
+variable "temporal_server" {
+  description = "Inputs for temporal-k8s charm module."
+  type = object({
+    app_name = string
+    channel  = string
+    revision = number
```

Comment:
Did you try with something like this, rather?

```
variable "temporal_server" {
  type = object({
    app_name           = optional(string, "temporal-server")
    channel           = optional(string, "1.23/stable")
    config             = optional(map(string), {})
    constraints        = optional(string, "arch=amd64")
    revision           = optional(number, null)
    units              = optional(number, 1)
  })
```

This way we have defaults for things like the channel and revision (null)